src/User32.Desktop/User32+SafeWindowStationHandle.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="SafeWindowStationHandle"/> nested type.
+    /// </content>
+    public static partial class User32
+    {
+        /// <summary>
+        /// Represents a Desktop handle that can be closed with <see cref="CloseWindowStation(IntPtr)"/>.
+        /// </summary>
+        public class SafeWindowStationHandle : SafeHandle
+        {
+            /// <summary>
+            /// A handle that may be used in place of <see cref="IntPtr.Zero"/>.
+            /// </summary>
+            public static readonly SafeWindowStationHandle Null = new SafeWindowStationHandle(IntPtr.Zero, false);
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SafeWindowStationHandle"/> class.
+            /// </summary>
+            public SafeWindowStationHandle()
+                : base(IntPtr.Zero, true)
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SafeWindowStationHandle"/> class.
+            /// </summary>
+            /// <param name="preexistingHandle">An object that represents the pre-existing handle to use.</param>
+            /// <param name="ownsHandle">
+            ///     <see langword="true" /> to have the native handle released when this safe handle is disposed or finalized;
+            ///     <see langword="false" /> otherwise.
+            /// </param>
+            public SafeWindowStationHandle(IntPtr preexistingHandle, bool ownsHandle = true)
+                : base(IntPtr.Zero, ownsHandle)
+            {
+                this.SetHandle(preexistingHandle);
+            }
+
+            public override bool IsInvalid => this.handle == IntPtr.Zero;
+
+            protected override bool ReleaseHandle()
+            {
+                if (this.IsInvalid)
+                {
+                    return false;
+                }
+
+                bool closed = CloseWindowStation(this.handle);
```

Comment:
The Kernel32 handles just call Close* here using that little lambda thingy. Is it not safer to do like I have done checking for validity first, being attentive of the return and if so setting the handle back to zero? I could change that behavior because I was a little annoyed by that ... (it could be a little too obsessive checking or playing too much on safety measures by my part though).